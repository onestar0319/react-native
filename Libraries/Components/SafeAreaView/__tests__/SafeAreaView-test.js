/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @format
 * @emails oncall+react_native
 * @flow strict-local
 */

'use strict';

const React = require('react');
const SafeAreaView = require('../SafeAreaView');
const ReactNativeTestTools = require('../../../Utilities/ReactNativeTestTools');
const View = require('../../View/View');
const Text = require('../../../Text/Text');

describe('<SafeAreaView />', () => {
  it('should render as expected', () => {
    ReactNativeTestTools.expectRendersMatchingSnapshot(
      'SafeAreaView',
      () => (
        <SafeAreaView>
          <View>
            <Text>Hello World!</Text>
          </View>
        </SafeAreaView>
      ),
      () => {
        jest.dontMock('../SafeAreaView');
      },
    );
  });
});
