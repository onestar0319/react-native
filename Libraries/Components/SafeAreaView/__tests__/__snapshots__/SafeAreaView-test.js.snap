// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SafeAreaView /> should render as <SafeAreaView> when mocked 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View>
    <Text>
      Hello World!
    </Text>
  </View>
</RCTSafeAreaView>
`;

exports[`<SafeAreaView /> should render as <SafeAreaView> when not mocked 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View>
    <Text>
      Hello World!
    </Text>
  </View>
</RCTSafeAreaView>
`;

exports[`<SafeAreaView /> should shallow render as <ForwardRef(SafeAreaView)> when mocked 1`] = `
<ForwardRef(SafeAreaView)>
  <View>
    <Text>
      Hello World!
    </Text>
  </View>
</ForwardRef(SafeAreaView)>
`;

exports[`<SafeAreaView /> should shallow render as <ForwardRef(SafeAreaView)> when not mocked 1`] = `
<ForwardRef(SafeAreaView)>
  <View>
    <Text>
      Hello World!
    </Text>
  </View>
</ForwardRef(SafeAreaView)>
`;
